konnectivity-client/pkg/client/client.go

```diff
@@ -141,7 +149,7 @@ func (t *grpcTunnel) serve(tunnelCtx context.Context, c clientConn) {
 
 	for {
 		pkt, err := t.stream.Recv()
-		if err == io.EOF {
+		if err == io.EOF || t.isClosing() {
```

Comment:
I've had problems with not draining connections before exiting. Can we ensure we do not need to drain the stream before exiting?

Reply:
From the documentation on the underlying gRPC method:

> 	// RecvMsg blocks until it receives a message into m or the stream is
	// done. It returns io.EOF when the stream completes successfully. On
	// any other error, the stream is aborted and the error contains the RPC
	// status.

I interpret that as EOF will only be returned if it doesn't return a different error (i.e. doesn't need to be drained). I also don't see us draining the stream anywhere else, nor do the gRPC docs mention "drain" anywhere.